src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1010,8 +1010,22 @@ private List<AssociativeNode> GetModifiedNodes(Subtree subtree, bool redefinitio
                     // It can then be handled normally regardless of its ForceExecution state
                     subtree.ForceExecution = false;
 
-                    //Check if the subtree (ie a node in graph) is an input and has primitive Right hand Node type
-                    if (redefinitionAllowed && st.IsInput && node is BinaryExpressionNode bne  && CoreUtils.IsPrimitiveASTNode(bne.RightNode))
+                    BinaryExpressionNode prevBNE = null;
+                    var bne = node as BinaryExpressionNode;
+                    if (bne != null){
+                        foreach (var astNode in st.AstNodes)
+                        {
+                            if(astNode is BinaryExpressionNode curBNE && curBNE.LeftNode is IdentifierNode id && id.Name == bne.LeftNode.Name)
```

Comment:
minor but I would still use Value for identifier node as it's specific to that node type, I think you missed my previous comment to the same effect

Reply:
should be addressed now.